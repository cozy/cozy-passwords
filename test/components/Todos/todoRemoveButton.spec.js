'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

// we import the not connected component version for testing
import { TodoRemoveButton } from 'components/Todos/TodoRemoveButton'
import Button from 'cozy-ui/react/Button'

// mock data for testing
import mockTodos from './_mocksTodos.json'

// async mock function
const mockDeleteTodo = jest.fn(() => Promise.resolve())
const todo = mockTodos[0]

describe('TodoRemoveButton component:', () => {
  beforeEach(() => {
    // reset all jest mock calls data before each test
    jest.resetAllMocks()
  })

  it('should be rendered correctly', () => {
    const component = shallow(
      <TodoRemoveButton todo={todo} deleteDocument={mockDeleteTodo} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should handle isWorking correctly (display spinner)', () => {
    const component = shallow(
      <TodoRemoveButton todo={todo} deleteDocument={mockDeleteTodo} />
    )
    expect(component.state().isWorking).toBe(false)
    component.setState({ isWorking: true })
    expect(component.getElement()).toMatchSnapshot()
  })

  it('should handle removeTodo correctly on button click', async () => {
    const component = shallow(
      <TodoRemoveButton todo={todo} deleteDocument={mockDeleteTodo} />
    )
    const removeButton = component.find(Button)
    await removeButton.simulate('click')
    expect(mockDeleteTodo.mock.calls.length).toBe(1)
    expect(mockDeleteTodo.mock.calls[0][0]).toBe(todo)
  })
})
