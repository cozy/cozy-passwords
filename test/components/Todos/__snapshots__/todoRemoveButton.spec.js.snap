// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoRemoveButton component: should be rendered correctly 1`] = `
<DefaultButton
  busy={false}
  className="todo-remove-button"
  disabled={false}
  extension="narrow"
  icon="trash"
  iconOnly={true}
  label="Delete"
  onClick={[Function]}
  theme="danger"
/>
`;

exports[`TodoRemoveButton component: should handle isWorking correctly (display spinner) 1`] = `
<DefaultButton
  busy={true}
  className="todo-remove-button"
  disabled={true}
  extension="narrow"
  icon="trash"
  iconOnly={true}
  label="Delete"
  onClick={[Function]}
  theme="danger"
/>
`;
